syntax = "proto3";
package waw.identity.v0;

message UserProfile {
  string id         = 1;
  string name       = 2;
  string email      = 3;
  string phone      = 4;
  string created_at = 5;
  string updated_at = 6;
}

message Empty {}
message ProfileDelta { UserProfile profile = 1; }

service IdentityService {
  rpc GetProfile(Empty) returns (UserProfile);
  rpc UpdateProfile(ProfileDelta) returns (UserProfile);
  rpc DeleteProfile(UserProfile) returns (Empty);
}
